tools/azure-pipelines-tasks-json-generate.mjs

```diff
@@ -0,0 +1,59 @@
+import os from 'os';
+import { promisify } from 'util';
+import fs from 'fs-extra';
+import g from 'glob';
+import JSON5 from 'json5';
+import shell from 'shelljs';
+import Git from 'simple-git';
+import path from 'upath';
+
+const glob = promisify(g);
+
+const localPath = path.join(os.tmpdir(), 'azure-pipelines-tasks');
+
+await (async () => {
+  await fs.ensureDir(localPath);
+  const git = Git(localPath);
+
+  if (await git.checkIsRepo()) {
+    await git.pull();
+  } else {
+    await git.clone(
+      'https://github.com/microsoft/azure-pipelines-tasks.git',
+      '.'
+    );
+  }
+
+  const files = (await glob(path.join(localPath, '**/task.json'))).map((file) =>
+    file.replace(`${localPath}/`, '')
+  );
+
+  /** @type {Record<string, Set<string>>} */
+  const tasks = {};
+
+  for (const file of files) {
+    const revs = (await git.raw(['rev-list', 'HEAD', '--', file])).split('\n');
```

Comment:
Can this be documented? I'm not sure what the purpose is. Is it just the content of the file locally, in which case why the need to use git?

Reply:
`git rev-list HEAD -- ${file}` gets a list of every commit in which the specific `task.json` was modified. `git show ${rev}:${file}` gets the content of that file at the specific commit. To summarize this script, it gets every revision of every `task.json` and extracts the version from it. Not every revision updates the version, and that's why I use a set to de-duplicate versions.

Reply:
I've added comments explaining how the script works, in case anyone ever needs to update it.